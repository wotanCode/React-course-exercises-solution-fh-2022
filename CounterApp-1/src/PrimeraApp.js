//import React, { Fragment } from "react";
import React from "react";
import PropTypes from "prop-types";

//FC funcional components
const PrimeraApp = ( {saludo, subtitulo } ) => {

  //const saludo = 'hola mundo'; //Valido
  // const saludo = 123; //Valido
  // const saludo = 123.2; //Valido
  // const saludo = true; //inValido
  // const saludo = [1,2,3,4,5] //Valido
  // const saludo = {
  //   nombre: 'hola',
  //   edad: 31,
  // }; //valido pero solo si usamos JSON.Stringify

  console.log(saludo)

  return (
    <>
      {/* <pre> { JSON.stringify(saludo, null, 3) } </pre> */}
      <h1>{saludo}</h1>
      <h2>{subtitulo}</h2>
    </>
  );
};

PrimeraApp.protoType = {
  saludo: PropTypes.string.isRequired,
}

PrimeraApp.defaultProps = {
  subtitulo: 'Soy un subtitulo',
}

export default PrimeraApp;