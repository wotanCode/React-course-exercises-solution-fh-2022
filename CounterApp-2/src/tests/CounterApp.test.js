import React from "react";
import PrimeraApp from "../CounterApp";
import {shallow} from 'enzyme'
import "@testing-library/jest-dom"

describe('Pruebas en componente counterApp', () => {

  let wrapper = shallow(<PrimeraApp />)

  beforeEach( () => {
    wrapper = shallow(<PrimeraApp />)
  })

  test('Prebas de integracion', () =>{

    const valor = '31';
    const wrapper = shallow(<PrimeraApp value={valor}/>)

    expect( wrapper ).toMatchSnapshot();
    
  });

  test('Pruebas de valor, debe retornar 100', () => {
    const valor = '100';
    const wrapper = shallow(<PrimeraApp value={valor}/>)

    const numeroValor = wrapper.find('h2').text().trim();

    expect(numeroValor).toBe(valor);
  });

  test('Debe de incrementar con +1', () =>{
    //const btn1 = wrapper.find('button').at(0).simulate('cliCk');
    //console.log(btn1)
    wrapper.find('button').at(0).simulate('click');
    const counterText = wrapper.find('h2').text().trim();
    //console.log(counterText);
    expect(counterText).toBe('11')
  });
  test('Debe de incrementar con -1', () =>{
    wrapper.find('button').at(2).simulate('click');
    const counterText = wrapper.find('h2').text().trim();
    expect(counterText).toBe('9')
  });

  test('Debe quedar en cero', () =>{
    const wrapper = shallow(<PrimeraApp value={105}/>)
    wrapper.find('button').at(0).simulate('click');
    wrapper.find('button').at(0).simulate('click');
    wrapper.find('button').at(1).simulate('click');
    const counterText = wrapper.find('h2').text().trim();
    
    expect(counterText).toBe('0')
  });
});