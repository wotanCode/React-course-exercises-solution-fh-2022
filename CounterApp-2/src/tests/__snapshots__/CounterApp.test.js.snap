// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente counterApp Prebas de integracion 1`] = `
<Fragment>
  <h1>
    Counter App
  </h1>
  <h2>
     
    31
     
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
